modules/engines/engine-xsodata/src/main/java/com/sap/xsk/xsodata/utils/XSKOData2EventHandlerUtils.java

```diff
@@ -0,0 +1,278 @@
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public class XSKOData2EventHandlerUtils {
+
+  public static final Logger LOGGER = LoggerFactory.getLogger(XSKScriptingOData2EventHandler.class);
+  public static final String ERROR_WHEN_PREPARING_TEMPORARY_TABLE_SQL = "Error when preparing temporary table SQL: ";
+  public static final String UNABLE_TO_HANDLE_BEFORE_CREATE_ENTITY_EVENT = "Unable to handle beforeCreateEntity event";
+  public static final String UNABLE_TO_HANDLE_AFTER_CREATE_ENTITY_EVENT = "Unable to handle afterCreateEntity event";
+  public static final String UNABLE_TO_HANDLE_ON_CREATE_ENTITY_EVENT = "Unable to handle onCreateEntity event";
+  public static final String UNABLE_TO_HANDLE_BEFORE_UPDATE_ENTITY_EVENT = "Unable to handle beforeUpdateEntity event";
+  public static final String UNABLE_TO_HANDLE_AFTER_UPDATE_ENTITY_EVENT = "Unable to handle afterUpdateEntity event";
+  public static final String UNABLE_TO_HANDLE_ON_UPDATE_ENTITY_EVENT = "Unable to handle onUpdateEntity event";
+  public static final String UNABLE_TO_HANDLE_BEFORE_DELETE_ENTITY_EVENT = "Unable to handle beforeDeleteEntity event";
+  public static final String UNABLE_TO_HANDLE_AFTER_DELETE_ENTITY_EVENT = "Unable to handle afterDeleteEntity event";
+  public static final String UNABLE_TO_HANDLE_ON_DELETE_ENTITY_EVENT = "Unable to handle onDeleteEntity event";
+  public static final String UNABLE_TO_DROP_TEMPORARY_TABLES = "Unable to drop temporary tables";
+  public static final String UNABLE_TO_CLOSE_CONNECTION = "Unable to close connection";
+
+  public static final String DUMMY_BUILDER = "dummyBuilder";
+  public static final String SELECT_BUILDER = "selectBuilder";
+  public static final String INSERT_BUILDER = "insertBuilder";
+  public static final String UPDATE_BUILDER = "updateBuilder";
+  public static final String SQL_CONTEXT = "sqlContext";
+
+  public static final String CONNECTION = "connection";
+  public static final String BEFORE_TABLE_NAME = "beforeTableName";
+  public static final String AFTER_TABLE_NAME = "afterTableName";
+  public static final String ON_CREATE_ENTITY_TABLE_NAME = "onCreateEntityTableName";
+  public static final String BEFORE_UPDATE_ENTITY_TABLE_NAME = "beforeUpdateEntityTableName";
+  public static final String BEFORE_DELETE_ENTITY_TABLE_NAME = "beforeDeleteEntityTableName";
+  public static final String ENTRY_MAP = "entryMap";
+  public static final String HANDLER = "handler";
+  public static final String ERROR_DOCUMENT = "errorDocument";
+
+  private static final String HTTP_STATUS_CODE = "HTTP_STATUS_CODE";
+  private static final String ERROR_MESSAGE = "ERROR_MESSAGE";
+  private static final String ERROR_DETAIL = "errordetail";
+
+  private static DataSource dataSource = (DataSource) StaticObjects.get(StaticObjects.DATASOURCE);
```

Comment:
DataSource or connection should be provided as input param to the method, otherwise you will have more than one db transaction for OData batch requests, which will be a bug